src/morphometrics/label/_tests/test_image_utils.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+import pytest
+
+from morphometrics.label.image_utils import expand_selected_labels
+
+
+def test_expand_selected_labels_2d():
```

Comment:
@haesleinhuepf , I am trying to run some tests on code that uses pyclesperanto. This test segfaults when running on github actions even though it passes locally. I have pasted the error below. If you have time, would you mind taking a look? Thanks!


```python
Current thread 0x00007f3eff317740 (most recent call first):
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pyclesperanto_prototype/_tier0/_device.py", line 101 in filter_devices
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pyclesperanto_prototype/_tier0/_device.py", line 72 in select_device
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pyclesperanto_prototype/_tier0/_device.py", line 43 in get_device
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pyclesperanto_prototype/_tier0/_pycl.py", line 69 in from_array
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pyclesperanto_prototype/_tier0/_opencl_backend.py", line 44 in from_array
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pyclesperanto_prototype/_tier0/_push.py", line 37 in push
  File "/home/runner/work/morphometrics/morphometrics/src/morphometrics/label/image_utils.py", line 49 in expand_selected_labels
  File "/home/runner/work/morphometrics/morphometrics/src/morphometrics/label/_tests/test_image_utils.py", line 14 in test_expand_selected_labels_2d
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/python.py", line 192 in pytest_pyfunc_call
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/python.py", line 1761 in runtest
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/runner.py", line 166 in pytest_runtest_call
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/runner.py", line 259 in <lambda>
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/runner.py", line 338 in from_call
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/runner.py", line 258 in call_runtest_hook
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/runner.py", line 219 in call_and_report
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/runner.py", line 130 in runtestprotocol
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/runner.py", line 111 in pytest_runtest_protocol
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/main.py", line 347 in pytest_runtestloop
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/main.py", line 322 in _main
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/main.py", line 268 in wrap_session
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/main.py", line 315 in pytest_cmdline_main
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_callers.py", line 39 in _multicall
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_manager.py", line 80 in _hookexec
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/pluggy/_hooks.py", line 265 in __call__
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/config/__init__.py", line 164 in main
  File "/usr/share/miniconda/envs/test/lib/python3.9/site-packages/_pytest/config/__init__.py", line 187 in console_main
  File "/usr/share/miniconda/envs/test/bin/pytest", line 8 in <module>
src/morphometrics/label/_tests/test_image_utils.py::test_expand_selected_labels_2d 
/home/runner/work/_temp/6b7326a1-98fc-4b59-9ff1-c1ea7e758c1b.sh: line 1:  4570 Segmentation fault      (core dumped) pytest -v --cov=./ --cov-report=xml
```

Reply:
This error location `pyclesperanto_prototype/_tier0/_device.py", line 101 in filter_devices` suggests an issue while [OpenCL device discovery](https://github.com/clEsperanto/pyclesperanto_prototype/blob/master/pyclesperanto_prototype/_tier0/_device.py#L101). The best opportunity to fix this might be the github-CI config, especially [this section](https://github.com/kevinyamauchi/morphometrics/blob/labeling-gui/.github/workflows/test_and_deploy.yml#L65-L68). 

Consider comparing the config with [other projects where the CI passes using clesperanto](https://github.com/haesleinhuepf/apoc/blob/main/.github/workflows/test_and_deploy.yml).

One more option could be to use [oclgrind](https://github.com/jrprice/Oclgrind) instead of [pocl](http://portablecl.org/). I must admit though, both have hickups in github CI.

Also a hint: I think we never managed to run pyopencl-based-tests on github/Windows. That's why the other projects using clesperanto don't test that system at all...

You can debug the available devices by running `clinfo` from the command line, and this should also work from within github CI.

One more idea: We recently saw this line helping on user's linux systems to make things work smoothly:
```
conda install -c conda-forge ocl-icd-system
```

Reply:
Thank you for the quick response and tips, @haesleinhuepf ! I will give these a go.

Reply:
Hey @kevinyamauchi ,

just FYI: I'm getting the same error now in [napari-apoc](https://github.com/haesleinhuepf/napari-accelerated-pixel-and-object-classification/runs/6638644766?check_suite_focus=true). Thus, it's reasonable that this is related to an upstream change, e.g. in pyopencl. I keep you posted.

Reply:
Thanks! I saw https://github.com/haesleinhuepf/apoc/pull/24. I can try that here and see if it fixes it.